api/v1alpha1/deployment_manager_types.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeploymentManager is the schema a deployment manager.
+//
+// +kubebuilder:resource:shortName=dm
```

Comment:
It is more descriptive, but it would make other short names longer and harder to type, for example `dmgrtemplate` instead of `dmtemplate`. I will keep this for now, unless you have a strong objection.